Flaky e2e test: Should add a task when CHW completes a task successfully, and that task creates another task

**Describe the issue**
The test fails with:
```
AssertionError: expected [ Element{ …(232) }, …(2), …(20) ] to have a length of 2 but got 3
    at Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/default/tasks/tasks.wdio-spec.js:93:26)
```

<details> <summary>Detail logs</summary>

```
RUNNING in chrome - file:///tests/e2e/default/tasks/tasks.wdio-spec.js
[0-29] (node:15121) Warning: Setting the NODE_TLS_REJECT_UNAUTHORIZED environment variable to '0' makes TLS connections and HTTPS requests insecure by disabling certificate verification.
[0-29] Error: Failed to sync
[0-29]     at syncAndWaitForSuccess (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:419:13)
[0-29]     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
[0-29]     at async Object.sync (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:445:3)
[0-29]     at async Object.updateSettings (/home/runner/work/cht-core/cht-core/tests/utils/index.js:600:5)
[0-29]     at async Object.compileTasks (/home/runner/work/cht-core/cht-core/tests/page-objects/default/tasks/tasks.wdio.page.js:99:3)
[0-29]     at async Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/default/tasks/tasks.wdio-spec.js:73:5)
[0-29]     at async Context.executeAsync (/home/runner/work/cht-core/cht-core/node_modules/@wdio/mocha-framework/node_modules/@wdio/utils/build/index.js:1030:20)
[0-29]     at async Context.testFrameworkFnWrapper (/home/runner/work/cht-core/cht-core/node_modules/@wdio/mocha-framework/node_modules/@wdio/utils/build/index.js:1102:14)
[0-29] AssertionError in "Tasks.should add a task when CHW completes a task successfully, and that task creates another task"
AssertionError: expected [ Element{ …(232) }, …(2), …(20) ] to have a length of 2 but got 3
    at Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/default/tasks/tasks.wdio-spec.js:93:26)
[0-29] Error: Failed to sync
[0-29]     at syncAndWaitForSuccess (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:419:13)
[0-29]     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
[0-29]     at async Object.sync (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:445:3)
[0-29]     at async Object.updateSettings (/home/runner/work/cht-core/cht-core/tests/utils/index.js:600:5)
[0-29]     at async Object.compileTasks (/home/runner/work/cht-core/cht-core/tests/page-objects/default/tasks/tasks.wdio.page.js:99:3)
[0-29]     at async Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/default/tasks/tasks.wdio-spec.js:107:5)
[0-29]     at async Context.executeAsync (/home/runner/work/cht-core/cht-core/node_modules/@wdio/mocha-framework/node_modules/@wdio/utils/build/index.js:1030:20)
[0-29]     at async Context.testFrameworkFnWrapper (/home/runner/work/cht-core/cht-core/node_modules/@wdio/mocha-framework/node_modules/@wdio/utils/build/index.js:1102:14)
[0-29] Error: waitUntil condition timed out after 10000ms
[0-29]     at <anonymous> (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/webdriverio/build/node.js:5706:13)
[0-29]     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
[0-29]     at async Browser.wrapCommandFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/@wdio/utils/build/index.js:907:23)
[0-29]     at async syncAndWaitForSuccess (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:408:5)
[0-29]     at async syncAndWaitForSuccess (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:423:12)
[0-29]     at async Object.sync (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:445:3)
[0-29]     at async Object.updateSettings (/home/runner/work/cht-core/cht-core/tests/utils/index.js:600:5)
[0-29]     at async Object.compileTasks (/home/runner/work/cht-core/cht-core/tests/page-objects/default/tasks/tasks.wdio.page.js:99:3)
[0-29]     at async Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/default/tasks/tasks.wdio-spec.js:107:5)
[0-29]     at async Context.executeAsync (/home/runner/work/cht-core/cht-core/node_modules/@wdio/mocha-framework/node_modules/@wdio/utils/build/index.js:1030:20)
[0-29] 2025-05-30T11:16:12.986Z ERROR webdriver: WebDriverError: element not interactable
[0-29]   (Session info: chrome=136.0.7103.92) when running "element/f.660619D5206769F0BDA2CD0B3DBA9196.d.12C7F20A5BAA4BC7BED9A0C400BF8718.e.1072/click" with method "POST"
[0-29] webdriverio(middleware): element did not become interactable: Element <span>Sync now</span> did not become interactable
[0-29]     at FetchRequest._request (file:///home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/webdriver/build/node.js:1803:19)
[0-29]     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
[0-29]     at async Element.wrapCommandFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/@wdio/utils/build/index.js:907:23)
[0-29]     at async Element.wrapCommandFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/@wdio/utils/build/index.js:907:23)
[0-29]     at async Element.elementErrorHandlerCallbackFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/webdriverio/build/node.js:8387:24)
[0-29]     at async Element.wrapCommandFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/@wdio/utils/build/index.js:907:23)
[0-29]     at async elementClick (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/webdriverio/build/node.js:6371:12)
[0-29]     at async Element.wrapCommandFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/@wdio/utils/build/index.js:907:23)
[0-29]     at async Element.elementErrorHandlerCallbackFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/webdriverio/build/node.js:8387:24)
[0-29]     at async Element.wrapCommandFn (/home/runner/work/cht-core/cht-core/node_modules/@wdio/runner/node_modules/@wdio/utils/build/index.js:907:23) {
[0-29]   url: URL {
[0-29]     href: 'http://localhost:42781/session/423a9b4a3ea2a95033bef6995690a4a7/element/f.660619D5206769F0BDA2CD0B3DBA9196.d.12C7F20A5BAA4BC7BED9A0C400BF8718.e.1072/click',
[0-29]     origin: 'http://localhost:42781',
[0-29]     protocol: 'http:',
[0-29]     username: '',
[0-29]     password: '',
[0-29]     host: 'localhost:42781',
[0-29]     hostname: 'localhost',
[0-29]     port: '42781',
[0-29]     pathname: '/session/423a9b4a3ea2a95033bef6995690a4a7/element/f.660619D5206769F0BDA2CD0B3DBA9196.d.12C7F20A5BAA4BC7BED9A0C400BF8718.e.1072/click',
[0-29]     search: '',
[0-29]     searchParams: URLSearchParams {},
[0-29]     hash: ''
[0-29]   },
[0-29]   opts: {
[0-29]     method: 'POST',
[0-29]     redirect: 'follow',
[0-29]     signal: AbortSignal { aborted: false },
[0-29]     body: {},
[0-29]     headers: Headers {
[0-29]       'Content-Type': 'application/json; charset=utf-8',
[0-29]       Connection: 'keep-alive',
[0-29]       Accept: 'application/json',
[0-29]       'User-Agent': 'webdriver/9.12.5',
[0-29]       'Content-Length': '2'
[0-29]     },
[0-29]     dispatcher: Agent {
[0-29]       _events: [Object: null prototype] {},
[0-29]       _eventsCount: 0,
[0-29]       _maxListeners: undefined,
[0-29]       [Symbol(shapeMode)]: false,
[0-29]       [Symbol(kCapture)]: false,
[0-29]       [Symbol(nodejs.stream.destroyed)]: false,
[0-29]       [Symbol(onDestroyed)]: null,
[0-29]       [Symbol(closed)]: false,
[0-29]       [Symbol(onClosed)]: [],
[0-29]       [Symbol(dispatch interceptors)]: [Array],
[0-29]       [Symbol(options)]: [Object],
[0-29]       [Symbol(maxRedirections)]: 0,
[0-29]       [Symbol(factory)]: [Function: defaultFactory],
[0-29]       [Symbol(clients)]: [Map],
[0-29]       [Symbol(onDrain)]: [Function (anonymous)],
[0-29]       [Symbol(onConnect)]: [Function (anonymous)],
[0-29]       [Symbol(onDisconnect)]: [Function (anonymous)],
[0-29]       [Symbol(onConnectionError)]: [Function (anonymous)],
[0-29]       [Symbol(Intercepted Dispatch)]: [Function: Intercept]
[0-29]     }
[0-29]   }
[0-29] }
[0-29] Error: Failed to sync
[0-29]     at syncAndWaitForSuccess (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:419:13)
[0-29]     at process.processTicksAndRejections (node:internal/process/task_queues:105:5)
[0-29]     at async syncAndWaitForSuccess (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:423:12)
[0-29]     at async Object.sync (/home/runner/work/cht-core/cht-core/tests/page-objects/default/common/common.wdio.page.js:445:3)
[0-29]     at async Object.updateSettings (/home/runner/work/cht-core/cht-core/tests/utils/index.js:600:5)
[0-29]     at async Object.compileTasks (/home/runner/work/cht-core/cht-core/tests/page-objects/default/tasks/tasks.wdio.page.js:99:3)
[0-29]     at async Context.<anonymous> (/home/runner/work/cht-core/cht-core/tests/e2e/default/tasks/tasks.wdio-spec.js:136:5)
[0-29]     at async Context.executeAsync (/home/runner/work/cht-core/cht-core/node_modules/@wdio/mocha-framework/node_modules/@wdio/utils/build/index.js:1030:20)
[0-29]     at async Context.testFrameworkFnWrapper (/home/runner/work/cht-core/cht-core/node_modules/@wdio/mocha-framework/node_modules/@wdio/utils/build/index.js:1102:14)
[0-29] FAILED in chrome - file:///tests/e2e/default/tasks/tasks.wdio-spec.js
```
</details> 

Example run:
https://github.com/medic/cht-core/actions/runs/15330930633/job/43162469921

--- a/tests/e2e/default/tasks/tasks.wdio-spec.js
+++ b/tests/e2e/default/tasks/tasks.wdio-spec.js
@@ -87,6 +87,7 @@ describe('Tasks', () => {
 
   it('should add a task when CHW completes a task successfully, and that task creates another task', async () => {
     await tasksPage.compileTasks('tasks-breadcrumbs-config.js', true);
+    await browser.pause(500);
 
     await commonPage.goToTasks();
     let list = await tasksPage.getTasks();
@@ -134,10 +135,9 @@ describe('Tasks', () => {
 
   it('Should show error message for bad config', async () => {
     await tasksPage.compileTasks('tasks-error-config.js', true);
+
     await commonPage.goToTasks();
-
     const { errorMessage, url, username, errorStack } = await commonPage.getErrorLog();
-
     expect(username).to.equal(chw.username);
     expect(url).to.equal('localhost');
     expect(errorMessage).to.equal('Error fetching tasks');

--- a/tests/e2e/default/tasks/tasks.wdio-spec.js
+++ b/tests/e2e/default/tasks/tasks.wdio-spec.js
@@ -86,7 +86,7 @@ describe('Tasks', () => {
   });
 
   it('should add a task when CHW completes a task successfully, and that task creates another task', async () => {
-    await tasksPage.compileTasks('tasks-breadcrumbs-config.js', false);
+    await tasksPage.compileTasks('tasks-breadcrumbs-config.js', true);
 
     await commonPage.goToTasks();
     let list = await tasksPage.getTasks();

--- a/tests/e2e/default/tasks/tasks.wdio-spec.js
+++ b/tests/e2e/default/tasks/tasks.wdio-spec.js
@@ -48,6 +48,16 @@ describe('Tasks', () => {
     reported_date: new Date().getTime(),
   });
 
+  const DOCS_TO_KEEP = [
+    ...Array.from(places.values()).map(doc => doc._id),
+    chwContact._id,
+    patient._id,
+    patient2._id,
+    'fixture:user:user1',
+    'org.couchdb.user:user1',
+    [/^form:/],
+  ];
+
   before(async () => {
     await utils.saveDocs([
       ...places.values(), healthCenter2, chwContact, patient, patient2,
@@ -66,7 +76,8 @@ describe('Tasks', () => {
 
   afterEach(async () => {
     await commonPage.logout();
-    await utils.revertSettings(true);
+    await utils.revertDb(DOCS_TO_KEEP, true);
+    await utils.revertSettings(false);
   });
 
   it('should remove task from list when CHW completes a task successfully', async () => {
@@ -90,7 +101,7 @@ describe('Tasks', () => {
 
     await commonPage.goToTasks();
     let list = await tasksPage.getTasks();
-    expect(list).to.have.length(2);
+    expect(list).to.have.length(3);
     let task = await tasksPage.getTaskByContactAndForm('Megan Spice', 'person_create');
     await task.click();
     await tasksPage.waitForTaskContentLoaded('Home Visit');
@@ -100,7 +111,7 @@ describe('Tasks', () => {
     await commonPage.sync({ expectReload: true });
     task = await tasksPage.getTaskByContactAndForm('Megan Spice', 'person_create_follow_up');
     list = await tasksPage.getTasks();
-    expect(list).to.have.length(3);
+    expect(list).to.have.length(4);
   });
 
   it('should load multiple pages of tasks on infinite scrolling', async () => {

--- a/tests/e2e/default/tasks/tasks.wdio-spec.js
+++ b/tests/e2e/default/tasks/tasks.wdio-spec.js
@@ -48,16 +48,6 @@ describe('Tasks', () => {
     reported_date: new Date().getTime(),
   });
 
-  const DOCS_TO_KEEP = [
-    ...Array.from(places.values()).map(doc => doc._id),
-    chwContact._id,
-    patient._id,
-    patient2._id,
-    'fixture:user:user1',
-    'org.couchdb.user:user1',
-    [/^form:/],
-  ];
-
   before(async () => {
     await utils.saveDocs([
       ...places.values(), healthCenter2, chwContact, patient, patient2,
@@ -76,8 +66,7 @@ describe('Tasks', () => {
 
   afterEach(async () => {
     await commonPage.logout();
-    await utils.revertDb(DOCS_TO_KEEP, true);
-    await utils.revertSettings(false);
+    await utils.revertSettings(true);
   });
 
   it('should remove task from list when CHW completes a task successfully', async () => {
@@ -101,7 +90,7 @@ describe('Tasks', () => {
 
     await commonPage.goToTasks();
     let list = await tasksPage.getTasks();
-    expect(list).to.have.length(3);
+    expect(list).to.have.length(2);
     let task = await tasksPage.getTaskByContactAndForm('Megan Spice', 'person_create');
     await task.click();
     await tasksPage.waitForTaskContentLoaded('Home Visit');
@@ -111,7 +100,7 @@ describe('Tasks', () => {
     await commonPage.sync({ expectReload: true });
     task = await tasksPage.getTaskByContactAndForm('Megan Spice', 'person_create_follow_up');
     list = await tasksPage.getTasks();
-    expect(list).to.have.length(4);
+    expect(list).to.have.length(3);
   });
 
   it('should load multiple pages of tasks on infinite scrolling', async () => {

